### ISSUE TITLE
add Runnable support to receiver builder

### ISSUE THREAD

**Issue by hepin1989** [akka/akka#25872]
```
    return receiveBuilder()
      .match(interestedEventClazz,
             event -> handleEvent(event))
      .matchEquals(MetricsLoggingTick.getInstance(),
                   tick -> doLogging())
      .build();
```
I would more like to write 

```
    return receiveBuilder()
      .match(interestedEventClazz,
             event -> handleEvent(event))
      .matchEquals(MetricsLoggingTick.getInstance(),
                   () -> doLogging())
      .build();
```

**Comment by hepin1989** [comment]
I will send a PR for this.
They are all not being used 
![image](https://user-images.githubusercontent.com/501740/49217338-40dde600-f408-11e8-84a7-058302e2732e.png)

**Comment by raboof** [comment]
(we forgot to close this one when we closed the PR, see #26007)
